npm/private/npm_translate_lock.bzl

```diff
@@ -521,6 +521,15 @@ WARNING: `package_json` attribute in `npm_translate_lock(name = "{name}")` is de
         lifecycle_hooks_execution_requirements,
     )
 
+    # The default `patch_args` of {"*": ["-p0"]} conflicts with the pnpm.patchedDependencies
+    # requirement of "-p1" for patches. When pnpm patches are used, it's highly likely that
+    # `patches` is unset, so when it is empty remove any strip prefixes from the default or
+    # any user-supplied argument as they won't have an effect anyway.
+    if len(patches) == 0 and patch_args:
```

Comment:
Would it be possible and any nicer changing the default to `patch_args = None` and adding the default when there are patches instead of clearing it when no patches? Mainly to avoid the extra strip method...

```
def npm_translate_lock(
        name,
        ....
        patches = {},
        patch_args = None,

...

    if len(patches) > 0 and patch_args == None:
       patch_args = {"*": ["-p0"]}
```

Reply:
My intention to perform the substitution whether `patch_args` is defaulted or the user supplies a value. As long as `patches` is empty, the strip prefix should be irrelevant because we always override the pnpm patches with `-p1`.

Reply:
Then they'd be overriding the `None` default and we wouldn't have to do anything though?

Reply:
I didn't want to default to `None` because then the stardocs would be inaccurate about what the default it.

Reply:
Isn't it even more inaccurate having the magic defaults depending on a combo of attributes and values?

Reply:
In general I would say so, however these changes to the defaults cannot have any effect if there are no manual patches supplied, and only helps to avoid a common error case. When there are only pnpm patches, any other strip prefix doesn't make sense, so it's harmless to remove it as we override it anyway.

Reply:
In the current state, someone who doesn't set `patches` or `patch_args` and just uses pnpm patches gets the error.

Reply:
@gregmagolan Did you have any opinions on this?

Reply:
Catching up on my `patch`-fu. It looks like by default (if you don't specify `-p0`) then `patch` will strip off all leading directories and just leave the base filename. That explains why we want `-p0` as the reasonable default.

How about calling `_remove_strip_prefix_arg` right before appending the `-p1` to `patch_args` if `pnpm_patched` is True in `npm/private/npm_translate_lock_generate.bzl` instead of doing it here?

Reply:
Can we avoid `_remove_strip_prefix_arg` completely though? I feel like it's odd (and dangerous) to modify a user-provided config in certain cases, it just leads to too many edge cases and hard to understand scenarios?

Reply:
That's what we do currently and it causes issues with the default `-p0` when using pnpm patches which prefer the `-p1`

Reply:
I mean change how our defaults work. Different defaults depending if we are using `patches` or assuming we're using pnpm (because `patches` is empty). Different defaults in these 2 cases instead of modifying/converting the args in 1/2 cases.

If that makes sense?

Reply:
It's hard to document in that case since stardoc wouldn't show the default.

I think it is cleaner to document that `-p1` is forced for any patches that are listed in the pnpm package.json patches and both the default and any user supplied `patch_args` for `-p` or `--strip` are ignored in that case.

Reply:
Can you think of a way of doing that without the current `_remove_strip_prefix_arg()` implementation though?

Scanning args for various `-p*` or `--strip=*` names and modifying the args is just something I'd prefer avoiding, especially with almost no test coverage :/

Reply:
I don't have any better ideas besides that. Maybe just appending `-p1` is sufficient if patch lets the last strip argument win but we'd have to test that.